quickwit/quickwit-ingest-api/src/ingest_api_service.rs

```diff
@@ -93,41 +91,63 @@ impl IngestApiService {
         })
     }
 
-    async fn ingest(
-        &mut self,
-        request: IngestRequest,
-        ctx: &ActorContext<Self>,
-    ) -> crate::Result<IngestResponse> {
-        // Check all indexes exist assuming existing queues always have a corresponding index.
-        let first_non_existing_queue_opt = request
-            .doc_batches
-            .iter()
-            .map(|batch| batch.index_id.as_str())
-            .find(|index_id| !self.queues.queue_exists(index_id));
+    pub fn partition_id(&self) -> &str {
+        &self.partition_id
+    }
+
+    pub async fn create_queue(&self, request: CreateQueueRequest) -> crate::Result<()> {
+        self.queues.create_queue(&request.queue_id).await?;
+        Ok(())
+    }
+
+    pub async fn create_queue_if_not_exists(
+        &self,
+        request: CreateQueueIfNotExistsRequest,
+    ) -> crate::Result<()> {
+        self.queues
+            .create_queue_if_not_exists(&request.queue_id)
+            .await?;
+        Ok(())
+    }
 
-        if let Some(index_id) = first_non_existing_queue_opt {
-            return Err(IngestServiceError::IndexNotFound {
-                index_id: index_id.to_string(),
-            });
+    pub async fn list_queues(
+        &self,
+        _request: ListQueuesRequest,
+    ) -> crate::Result<ListQueuesResponse> {
+        self.queues.list_queues().await
+    }
+
+    pub async fn drop_queue(&self, request: DropQueueRequest) -> crate::Result<()> {
+        self.queues.drop_queue(&request.queue_id).await?;
+        Ok(())
+    }
+
+    async fn ingest_inner(&self, request: IngestRequest) -> crate::Result<IngestResponse> {
+        // Check all indexes exist assuming existing queues always have a corresponding index.
+        for doc_batch in &request.doc_batches {
+            if !self.queues.queue_exists(&doc_batch.index_id).await {
```

Comment:
We have a race condition now... If someone deletes the queues after the existence check, the ingest request can fail partially.

Reply:
Alternatively we can just properly identify & accept the new behavior of this function...